### PR TITLE
[FLOC-4354] Remove container agent and container API docs

### PR DESCRIPTION
**Pull request by wallrj**
Fixes: https://clusterhq.atlassian.net/browse/FLOC-4354

The updated documentation can be viewed at:
 * http://clusterhq-staging-docs.s3-website-us-east-1.amazonaws.com/remove-container-agent-docs-FLOC-4354/

### CODE REVIEW

File: docs/supported/flockercontainers.rst [hunk 1]
```diff
@@ -8,20 +8,16 @@ Running Flocker in Containers
 
     <div class="admonition labs">
         <p>This page describes one of our experimental projects, developed to less rigorous quality and testing standards than the mainline Flocker distribution. It is not built with production-readiness in mind.</p>
```
Comment (wallnerryan) [inline docs/supported/flockercontainers.rst]: This isnt part of our release process to update the containers, but I feel could be integrated. E.g. this documentation is probably out of date in the sense that the containers for 1.14 , 1.15 havent been built I think, i usually kick these off, last I did was fro christian at jetstack around 1.13 I think.

**Comment by wallnerryan** [comment]
http://ci-live.clusterhq.com:8080/job/ClusterHQ-flocker/job/remove-container-agent-docs-FLOC-4354/job/run_acceptance_loopback_on_AWS_Ubuntu_Trusty_for_flocker.acceptance/3/
loopback on aws seems to be having an issue with the docker daemon?
`Cannot connect to the Docker daemon. Is the docker daemon running on this host?`

- http://ci-live.clusterhq.com:8080/job/ClusterHQ-flocker/job/remove-container-agent-docs-FLOC-4354/job/_run_acceptance_on_Rackspace_Ubuntu_Trusty_with_Cinder/1/
Wamp, time to refresh the build slave or clean up for some room.
`IOError: [Errno 28] No space left on device`

Same here with
` http://ci-live.clusterhq.com:8080/job/ClusterHQ-flocker/job/remove-container-agent-docs-FLOC-4354/job/_run_acceptance_on_GCE_Ubuntu_Xenial_with_GCE/1/
`IOError: [Errno 28] No space left on device`

File: docs/supported/flockercontainers.rst (hunk 1 above)
Comment (wallnerryan) [inline docs/supported/flockercontainers.rst]: Or even if we need these at all anymore

**Comment by wallnerryan** [comment]
only reason i used changes requested, which looks like a hard denial on github im realizing (lol) is the jenkins tests, so will approve once they run.

File: docs/supported/flockercontainers.rst (hunk 1 above)
Comment (wallrj) [inline docs/supported/flockercontainers.rst]: Sooner or later we're going to have to create an official Flocker Docker image and at that time we can update the docs.

**Comment by wallnerryan** [comment]
Just saw that https://clusterhq.atlassian.net/browse/FLOC-4506 removes the tests, why do I not see failures in this branch if it removes the container agent scripts. Possibly because jenkins sitll had container agent installed?

File: docs/supported/flockercontainers.rst (hunk 1 above)
Comment (wallnerryan) [inline docs/supported/flockercontainers.rst]: 👍

**Comment by wallrj** [comment]
In https://github.com/ClusterHQ/flocker/pull/2913#issuecomment-250751959 @wallnerryan wrote:

> loopback on aws seems to be having an issue with the docker daemon?
> Cannot connect to the Docker daemon. Is the docker daemon running on this host?

I've restarted that test. 
That test is run on an on-demand build server which has since been deleted.
So not sure what the docker problem was. 
Let's see if it happens again.


>  Wamp, time to refresh the build slave or clean up for some room.
>  IOError: [Errno 28] No space left on device

Yep, the problem was that those build servers are now downloading a new version of our package build Docker image every night which had filled >40GiB of disk space.
I deleted them and restarted the builds.

**Comment by wallrj** [comment]
In https://github.com/ClusterHQ/flocker/pull/2913#issuecomment-250756104 @wallnerryan  wrote:
> Just saw that https://clusterhq.atlassian.net/browse/FLOC-4506 removes the tests, why do I not see failures in this branch if it removes the container agent scripts. Possibly because jenkins sitll had container agent installed?

The `flocker-container-agent` command is still installed, this  branch only removes the initial enable / start of that script when the cluster is provisioned.
Those acceptance tests are decorated with `@require_cluster(..., container_agent=True)` which ensures that ``flocker-container-agen`` is enabled and started on all the nodes before the test runs.
In the followup branch I remove those tests and the parameter to `require_cluster`.
Hope that makes sense.

**Comment by wallnerryan** [comment]
yup makes sense 👍

**Comment by wallnerryan** [comment]
> Yep, the problem was that those build servers are now downloading a new version of our package build Docker image every night which had filled >40GiB of disk space.
I deleted them and restarted the builds.

May i suggest... we setup some cron job or something like the below?
`docker images | grep "weeks ago" | awk '{print $3}' | xargs docker rmi`

**Pull request merged**